### PR TITLE
GH actions

### PR DESCRIPTION
**Pull request by keynmol**
- [x] verify that build and test works
- [x] verify that if condition on publish artifacts works
- [x] ~verify what changes are needed to make site release script work with GITHUB_TOKEN instead~
      - Deploy key should work, just need to change enough settings (I've verified it works in my fork)

### CODE REVIEW

File: project/WeaverPlugin.scala [hunk 1]
```diff
@@ -395,4 +396,74 @@ object WeaverPlugin extends AutoPlugin {
         .toSeq
   )
 
+  def createBuildCommands(projects: Seq[ProjectReference]) = {
```
Comment (keynmol) [inline project/WeaverPlugin.scala:399]: I will make it my life's purpose to find a way to do this better.

File: modules/framework/cats/test/src/MemoisedResourceTests.scala [hunk 2]
```diff
@@ -23,7 +23,7 @@ object MemoisedResourceTests extends SimpleIOSuite {
       resource =
         Resource.make(initialised.update(_ + 1))(_ => finalised.update(_ + 1))
       use <- MemoisedResource(resource).map(r =>
-        r.use(_ => IO.sleep(100.millis) *> used.update(_ + 1)))
+        r.use(_ => IO.sleep(250.millis) *> used.update(_ + 1)))
```
Comment (keynmol) [inline modules/framework/cats/test/src/MemoisedResourceTests.scala:26]: "Fixes" flaky test

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        env:
+          PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
+
+  site:
+    name: Publish Site
+    needs: [publish]
+    if: startsWith(github.ref, 'refs/tags/v')
```
Comment (keynmol) [inline .github/workflows/ci.yml:99]: only publish site on tags

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,122 @@
+name: CI
+
+on:
+  pull_request:
+    branches: ["*"]
+  push:
+    branches: ["master"]
+    tags: ["v*"]
+
+# env:
```
Comment (keynmol) [inline .github/workflows/ci.yml:10]: TODO: remove (we don't need as much memory it seems given that we're running jobs in parallel)

File: scripts/releaseSite.sc [hunk 5]
```diff
@@ -158,8 +161,9 @@ val installSSHScript: String =
 
 def doInstallSSH() =  {
   val tmp = os.temp(contents = installSSHScript,
+                    dir = os.pwd,
                     prefix = "docusaurus",
-                    suffix = "install_ssh.sh",
-                    perms = Set(OWNER_EXECUTE).asJava)
```
Comment (keynmol) [inline scripts/releaseSite.sc:163]: This was throwing exceptions on GHA trying to execute the file

File: scripts/releaseSite.sc [hunk 6]
```diff
@@ -158,8 +161,9 @@ val installSSHScript: String =
...
-  os.proc(tmp).call(cwd = os.pwd)
+                    suffix = "install_ssh.sh")
+  os.proc.apply(Seq("chmod", "+x", tmp.toString))
+  os.proc.apply(Seq("bash", tmp.toString)).call(cwd = os.pwd)
```
Comment (keynmol) [inline scripts/releaseSite.sc:168]: I had to shell out with the script, as executing it with `./...sh.` didn't work

**Review by Baccata — APPROVED** [review]
Blindly putting my life in your hands 😄

**Pull request merged**